recipes/cuda-cccl/meta.yaml

```diff
@@ -0,0 +1,63 @@
+{% set name = "cuda-cccl" %}
+{% set version = "12.0.90" %}
+{% set impl_version = "2.0.1" %}
+{% set platform = "linux-x86_64" %}  # [linux64]
+{% set platform = "linux-ppc64le" %}  # [ppc64le]
+{% set platform = "linux-sbsa" %}  # [aarch64]
+{% set platform = "windows-x86_64" %}  # [win]
+{% set extension = "tar.xz" %}  # [not win]
+{% set extension = "zip" %}  # [win]
+
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://developer.download.nvidia.com/compute/cuda/redist/cuda_cccl/{{ platform }}/cuda_cccl-{{ platform }}-{{ version }}-archive.{{ extension }}
+  sha256: 37f6e03b6f95579d8e2e96b9ba15eeb1224dcd18b21ad50c3ac249e408152a24  # [linux64]
+  sha256: c6de67acd6a65b24db0383f6330b842ed19b6c27a27a37f828e9f3e96cc96f72  # [ppc64le]
+  sha256: f2a19927c558a9c872fa87eda4c0dc0446e5c4c744df4f846d01f4d4e08c8cd7  # [aarch64]
+  sha256: dd422d4cc8462d112411659a56cc78cb6010d704f83db3e48c55a3f66eed5ec2  # [win]
+
+build:
+  number: 0
+  skip: true  # [osx]
+
+outputs:
+  - name: cuda-cccl
+    requirements:
+      run:
+        - cuda-cccl_{{ target_platform }} =={{ version }}
+        - cuda-cccl-impl =={{ impl_version }}
```

Comment:
I'm a little confused why `cuda-cccl` needs to depend on `cuda-cccl-impl` at all, because it appears to be fetching the CCCL from the CUDA Toolkit Archive? Won't that mean that both `cuda-cccl` and `cuda-cccl-impl` distribute files that would clobber each other?

Reply:
I think there's no clobbering. `cuda-cccl_{{ target_platform }}` installs to `$PREFIX/targets` and `cuda-cccl-impl` installs to $PREFIX`. 

That said, I am sensing people have different interpretations on the proposed resolution here. Even I am no longer sure whether my above question on `cuda-version` makes sense. Perhaps we should clean up the proposal and write up a short, up-to-date summary...

Reply:
I thought the idea was if someone wanted the `cuda-cccl-impl` that shipped with CTK then `cuda-cccl` would be the way to do that.

I put a summary of my understanding of the proposal in [an above comment](https://github.com/conda-forge/staged-recipes/pull/21953#issuecomment-1486960066) if that helps.

Reply:
Does `cuda-cccl` need to depend on `cuda-cccl-impl =={{ impl_version }}`? It seems it would prevent users from combining compilers or other packages depending on `cuda-version=12.0` with some later version of `cuda-cccl-impl`. The ability to install newer CCCL versions when building libraries with a fixed CUDA Toolkit version is important for downstream applications.

Reply:
I would expect other CTK packages, namely `cuda-cudart-dev_{{ target_platform }}`, to depend on `cuda-cccl_{{ target_platform }}`, not `cuda-cccl`. So a user would only lock `cuda-cccl-impl` to a CTK version if they install `cuda-cccl` explicitly. I [originally ](https://github.com/conda-forge/staged-recipes/pull/21953#discussion_r1150777350) had this as `>=`, but that would it more difficult to lock `cuda-cccl-impl` to a CTK version if that is intended.

Reply:
Fantastic. That's a good solution. I apologize, I think you might have said this before and I forgot that.